### PR TITLE
Feature/151

### LINKED ISSUE #151: Modal styling changes
Need a couple styling changes to the example types default modal to account for the fact that we get some pretty unruly content in there

* set a `px` max-width instead of `fit-content` since that's experimental and isn't working with the crazy nested tables that come through
* `overflow-x: scroll`
* change the current padding to a border so x-scrollable content scrolls within the visible whitespace instead of over it

### PR DESCRIPTION
**Pull request by doug-patterson**
Fixes #151 
Fixes https://github.com/smartprocure/bid-search/issues/1067

* styling changes to make the default modal better handle un-cooperative content

### CODE REVIEW

**Comment by decrapifier** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>Please assign someone to merge this PR, and optionally include people who should review.</td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>You should add at least 1 more reviewer to the PR</td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>The README has not been updated. Please update the README.</td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>Could not find any browser results.</td>
    </tr>
  </tbody>
</table>


<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 404b05472670035bc2408728dfc28d5d5a6bfbb9
</p>

File: src/layout/Modal.js [hunk 1]
```diff
@@ -25,10 +25,11 @@ let Modal = ({ isOpen, children, style = {} }) => (
           style={{
             backgroundColor: '#fff',
             borderRadius: 5,
-            maxWidth: 'fit-content',
+            maxWidth: 640,
+            overflowX: 'scroll',
             // minHeight: 300,
             margin: '0 auto',
-            padding: 30,
+            border: '30px solid white',
```
Comment (doug-patterson) [inline src/layout/Modal.js]: @daedalus28 we could also make the component take an `innerStyle` prop for this div and pass this through from grey vest

File: src/layout/Modal.js [hunk 2]
```diff
@@ -25,10 +25,11 @@ let Modal = ({ isOpen, children, style = {} }) => (
           style={{
             backgroundColor: '#fff',
             borderRadius: 5,
-            maxWidth: 'fit-content',
+            maxWidth: 640,
+            overflowX: 'scroll',
             // minHeight: 300,
             margin: '0 auto',
-            padding: 30,
```
Comment (doug-patterson) [inline src/layout/Modal.js]: changing from padding to border as this means the scrollable x overflow won't overlap the visible whitespace

File: src/layout/Modal.js (hunk 1 above)
Comment (daedalus28) [reply src/layout/Modal.js]: We should definitely do that. Also, the 30px border is super weird 🤔

Comment (daedalus28) [reply src/layout/Modal.js]: And if that's too tricky, we can just put a class on it and override in grey vest with css - but the 30px should not be in the core modal code

**Comment by doug-patterson** [comment]
@daedalus28 updated to add a couple classes to the default modal markup and then use those from the theme to style the modal. I've kept the 30px border as it's the best way I know of to get the desired effect (side scrolling inside of whitespace within a single element) -- to do it without that css hack would require nesting a second div (which would be fine if we prefer it that way, but I don't really see an issue with using the border for this purpose)

File: src/layout/Modal.js [hunk 3]
```diff
@@ -20,18 +20,20 @@ let Modal = ({ isOpen, children, style = {} }) => (
...
         }}
         onClick={F.off(isOpen)}
+        className="default-modal-bg"
       >
         <div
           style={{
             backgroundColor: '#fff',
             borderRadius: 5,
             maxWidth: 'fit-content',
+            padding: 30,
```
Comment (daedalus28) [inline src/layout/Modal.js]: let's just kill the default padding, border radius, color, etc that really belongs in grey vest (and so we don't need !important)

**Comment by doug-patterson** [comment]
@daedalus28 updated

File: src/themes/greyVest.js [hunk 4]
```diff
@@ -666,6 +666,14 @@ export let GVStyle = () => (
         padding: 15px;
         width: 400px !important;
       }
+
+      /* Modals */
+      .default-modal-wrap {
+        width: auto;
+        max-width: 640px!important;
```
Comment (daedalus28) [inline src/themes/greyVest.js]: Do we still need `!important`?

**Comment by doug-patterson** [comment]
@daedalus28 updated

**Pull request merged**